slack_bolt/context/complete/async_complete.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Dict, Optional
+
+from slack_sdk.web.async_client import AsyncWebClient
+from slack_sdk.web.async_slack_response import AsyncSlackResponse
+
+
+class AsyncComplete:
+    client: AsyncWebClient
+    function_execution_id: Optional[str]
+
+    def __init__(
+        self,
+        client: AsyncWebClient,
+        function_execution_id: Optional[str],
+    ):
+        self.client = client
+        self.function_execution_id = function_execution_id
+
+    async def __call__(self, outputs: Dict[str, Any] = {}) -> AsyncSlackResponse:
+        """Signal the successful completion of the custom function.
+
+        Kwargs:
+            outputs: Json serializable object containing the output values
+
+        Returns:
+            SlackResponse: The response object returned from slack
+
+        Raises:
+            ValueError: If this function cannot be used.
+        """
+        if self.function_execution_id is None:
+            raise ValueError("complete is unsupported here as there is no function_execution_id")
```

Comment:
If I understand this correctly, a non-`function_executed` event handler trying to invoke `complete` or `fail` will raise an exception.
Perhaps we should warn, instead, and `return` or `pass` the `__call__` so as to not cause runtime issues for the app? WDYT?

EDIT: huh, I guess other similar convenience functions like `respond`, which only work if the event payload includes a `response_url`, work like this (by raising an exception). Seems odd to me but I guess we should keep it consistent.